pkg/engine/engine.go

```diff
@@ -399,3 +411,17 @@ func isTemplateValid(ch *chart.Chart, templateName string) bool {
 func isLibraryChart(c *chart.Chart) bool {
 	return strings.EqualFold(c.Metadata.Type, "library")
 }
+
+func readTemplateMagicComments(templateBody string) templateOpts {
+	templateOpts := templateOpts{}
+	matches := magicCommentsRegexp.FindAllSubmatch([]byte(templateBody), -1)
+	for _, match := range matches {
+		if strings.EqualFold(string(match[1]), "delim") {
+			delim := strings.SplitN(string(match[2]), ",", 2)
+			templateOpts.delimL = strings.Repeat(delim[0], 2)
```

Comment:
Why can't I specify my own delimiter syntax? For example,

```
# helm: delim={%,%}
```

Why does it have to be a single character repeated twice? Why not allow single character delimiters?

```
# helm: delim=$(,)
```

What happens if I wanted commas as a delimiter?

```
# helm: delim=,,,
```

Reply:
What happens if I wrap the delimiters in single or double quotes?

```
# helm: delim="{,}"
```

or

```
# helm: "delim={,}"
```

Reply:
> Why can't I specify my own delimiter syntax? For example,
> 
> ```
> # helm: delim={%,%}
> ```
> 
> Why does it have to be a single character repeated twice? Why not allow single character delimiters?
> 
> ```
> # helm: delim=$(,)
> ```

I mentioned this in the PR description.
The problem is that when the template is rendered the line `# helm: delim={%,%}` is parsed by the engine and it will try to evaluate `{%,%}` and panic since `,` is not a valid `go-template` expression.

I could remove the magic comment from the template, but I think it adds value to keep it there since you can see the magic comments that were part of the rendering process in the final output which contributes to debugging, not to mention performance (altering the string templates, that are kept in memory)

I could also try to escape it somehow, but just the thought of it gives me a headache considering we are talking about a dynamic delimiter :-D 
```
# helm: delim={%,%}
```
should become
```
# helm: delim={%`{%`%},{%`%}`%}
```

> What happens if I wanted commas as a delimiter?
> 
> ```
> # helm: delim=,,,
> ```

😱

Maybe we can even help downstream users and limit the possible delimiters to a set of characters that we know will not collide with other things in the templates.
Imagine if users would choose "`" as the delimiter or even `#`...

What do you think about the set: `[ { ( %`?

This paves the way to another good option which is to have a closed set of "delimiting style":

```
# helm: delimiterStyle=curlyBrackets
# helm: delimiterStyle=squareBrackets
# helm: delimiterStyle=roundBrackets
# helm: delimiterStyle=percentageSigns
```

This will give users less freedom but in return will have a much more simple, stable and stupid-proof UX, as an added bonus this solves the problem of having the explicit delimiter in the template (the one explained in the top of this comment) because it's not using the delimiter itself to describe the delimiter.

Thoughts?

Reply:
I still think this would be better solved by implementing something similar to what was asked in https://github.com/helm/helm/issues/9302. We've heard more users ask about whether they can use alternative template engines like jinja2 rather than extending the existing `text/template` engine. See also https://github.com/helm/helm/issues/8290.

If we're going to allow users to extend the existing template engine, IMO it would be m,ore preferable to implement engine swap support rather than just switching config flags through a "magic comments" system. Then you can write your own template engine that extends Helm's current `pkg/engine`, swapping out the delimiter with your own choosing. If we find that system to be too clumsy then we can revisit this idea in the future. But we should be able to accomodate both your proposal here as well as what others have asked for in one fell swoop.

Reply:
I agree. I'd be extremely happy if Helm would have a pluggable template engine feature.
I don't see why this is a reason not to accept a tiny feature in the current built-in `text/template` engine.

Reply:
Fixed.
The new code will now parse `# helm: delimiter={%,%}` as valid input and will not "double" the delimiters.
It should also not panic for bad input